atc/engine/check_delegate.go

```diff
@@ -111,25 +118,56 @@ func (d *checkDelegate) WaitToRun(ctx context.Context, scope db.ResourceConfigSc
 		}
 	}
 
-	end, err := scope.LastCheckEndTime()
-	if err != nil {
-		if releaseErr := lock.Release(); releaseErr != nil {
-			logger.Error("failed-to-release-lock", releaseErr)
+	shouldRun := false
+	if !d.plan.IsPeriodic() {
+		lastCheckEndTime, lastCheckSucceeded, err := scope.LastCheckEndTime()
+		if err != nil {
+			if releaseErr := lock.Release(); releaseErr != nil {
+				logger.Error("failed-to-release-lock", releaseErr)
+			}
+			return nil, false, fmt.Errorf("get last check end time: %w", err)
 		}
 
-		return nil, false, fmt.Errorf("get last check end time: %w", err)
-	}
+		if !lastCheckSucceeded || lastCheckEndTime.Before(d.build.StartTime()) {
+			shouldRun = true
+		} else {
+			_, found, err := scope.LatestVersion()
```

Comment:
so: if the last check succeeded *after* this build started *and* it didn't produce a version, run another check.

this seems like a pretty specific scenario to ensure we run an extra check (vs. just assuming there's still no version) - did something break without this?

Reply:
Nothing broken without this. Maybe I over worried. I think this should only protect the case of manually cleaning up `resource_config_versions` table. I don't know if any users do so. For us, we only cleanup versions of `time` resources. I just removed the last version check.